light-client/src/header/eth_header.rs

```diff
@@ -236,45 +270,44 @@ impl ETHHeader {
             {
                 return Err(Error::UnexpectedVoteLength(self.extra_data.len()));
             }
-            let start = EXTRA_VANITY + VALIDATOR_NUM_SIZE + (num * VALIDATOR_BYTES_LENGTH);
+            let start = EXTRA_VANITY
+                + VALIDATOR_NUM_SIZE
+                + (num * VALIDATOR_BYTES_LENGTH)
+                + TURN_LENGTH_SIZE;
             let end = self.extra_data.len() - EXTRA_SEAL;
             &self.extra_data[start..end]
         };
 
         Rlp::new(attestation_bytes).try_into()
     }
 
-    // https://github.com/bnb-chain/bsc/blob/33e6f840d25edb95385d23d284846955327b0fcd/consensus/parlia/parlia.go#L342
-    pub fn get_validator_bytes(&self) -> Option<Validators> {
-        if self.extra_data.len() <= EXTRA_VANITY + EXTRA_SEAL {
-            return None;
-        }
-        let num = self.extra_data[EXTRA_VANITY] as usize;
-        if num == 0
-            || self.extra_data.len() <= EXTRA_VANITY + EXTRA_SEAL + num * VALIDATOR_BYTES_LENGTH
-        {
-            return None;
-        }
-        let start = EXTRA_VANITY + VALIDATOR_NUM_SIZE;
-        let end = start + num * VALIDATOR_BYTES_LENGTH;
-        Some(
-            self.extra_data[start..end]
-                .chunks(VALIDATOR_BYTES_LENGTH)
-                .map(|s| s.into())
-                .collect(),
-        )
+    pub fn is_epoch(&self) -> bool {
+        self.number % BLOCKS_PER_EPOCH == 0
     }
+}
 
-    pub fn get_validator_set(&self) -> Result<ValidatorSet, Error> {
-        Ok(self
-            .get_validator_bytes()
-            .ok_or_else(|| Error::MissingValidatorInEpochBlock(self.number))?
-            .into())
+// https://github.com/bnb-chain/bsc/blob/33e6f840d25edb95385d23d284846955327b0fcd/consensus/parlia/parlia.go#L342
+pub fn get_validator_bytes_and_tern_term(extra_data: &[u8]) -> Result<(Validators, u8), Error> {
+    if extra_data.len() <= EXTRA_VANITY + EXTRA_SEAL {
+        return Err(Error::UnexpectedExtraDataLength(extra_data.len()));
     }
-
-    pub fn is_epoch(&self) -> bool {
-        self.number % BLOCKS_PER_EPOCH == 0
+    let num = extra_data[EXTRA_VANITY] as usize;
+    if num == 0 || extra_data.len() <= EXTRA_VANITY + EXTRA_SEAL + num * VALIDATOR_BYTES_LENGTH {
+        return Err(Error::UnexpectedExtraDataLength(extra_data.len()));
     }
+    let start = EXTRA_VANITY + VALIDATOR_NUM_SIZE;
+    let end = start + num * VALIDATOR_BYTES_LENGTH;
+    let turn_length = extra_data[end];
+    if !(turn_length == 1 || (3..=9).contains(&turn_length)) {
```

Comment:
Only 1 and 3~9 are allowed according to 424-governable-number-of-consecutive-blocks